proposals/lip-0047.md

```diff
@@ -498,11 +498,147 @@ This module does not define any specific logic for other modules.
 
 This module does not define any endpoint for off-chain services.
 
-### Protocol Logic During Block Lifecycle
+### Genesis Block Processing
 
-#### After Genesis Block Execution
+The following two steps are executed as part of the genesis block processing, see the “Update genesis block schema and processing” LIP for details. 
 
-After the genesis block `g` is executed, the following logic is executed:
+#### Genesis State Initialization
+
+To initialize the state of a genesis block `g`, the following logic is executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the bytes included in the block assets for the PoA module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisPoAStoreSchema` schema, given below, then: 
+
+* Checks for the `genesisBlockAssetObject.validatorSubstore`: 
+  * Check that the `address` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array is distinct pairwise.
+  * Check that the `name` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array contains only characters from the set `[a-z0-9!@$&_.]`, is not empty and is at most `MAX_LENGTH_NAME` characters long.
+  * Check that the `generatorKey` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array has length 32.
+  * Check that the `blsKey` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array has length 48.
+  * Check that the `proofOfPossession` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array has length 96.
+
+* Checks for the `genesisBlockAssetObject.snapshotSubstore`: 
```

Comment:
Wouldn't it be easier here and also better encourage code re-use to reference the same validation logic already described above:
- For `genesisBlockAssetObject.snapshotSubstore.activeValidators` apply the same verifications as for `trs.params.newValidators` in the update authority command verification.
- For `genesisBlockAssetObject.snapshotSubstore.threshold` apply the same verifications as for `trs.params.threshold` in the update authority command verification.

Reply:
I think in this case it is a bit dangerous to refer to that logic since for the update authority command verification the entries in the validator substore already exist and we do not need to check for uniqueness etc.

Reply:
Okay, can be resolved.